File: app/Models/Category.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Concerns\HasUuids;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Category extends Model
{
    use HasUuids, SoftDeletes;

    protected $table = 'categories';

    protected $fillable = [
        'parent_id',
        'name',
        'slug',
        'description',
        'image_url',
        'is_visible',
        'sort_order',
        'age_restricted',
        'no_return',
    ];

    protected $casts = [
        'is_visible'  => 'boolean',
        'age_restricted' => 'boolean',
        'no_return'   => 'boolean',
        'sort_order'  => 'integer',
        'created_at'  => 'datetime',
        'updated_at'  => 'datetime',
        'deleted_at'  => 'datetime',
    ];

    public function parent()
    {
        return $this->belongsTo(Category::class, 'parent_id');
    }

    public function children()
    {
        return $this->hasMany(Category::class, 'parent_id')->orderBy('sort_order')->orderBy('name');
    }

    public function products()
    {
        return $this->hasMany(Product::class, 'category_id');
    }
}

File: mobile/lib/models/category.dart
/// Категория товаров (GET /widget/categories).
class Category {
  final String id;
  final String name;
  final String slug;
  final int sortOrder;

  /// 18+: перед показом товаров этой категории спрашиваем возраст, фото
  /// блюрятся до подтверждения (см. AgeGate).
  final bool ageRestricted;

  /// Товары этой категории возврату не подлежат — бейдж на карточке.
  final bool noReturn;

  final List<Category> children;

  const Category({
    required this.id,
    required this.name,
    required this.slug,
    required this.sortOrder,
    this.ageRestricted = false,
    this.noReturn = false,
    this.children = const [],
  });

  factory Category.fromJson(Map<String, dynamic> json) => Category(
    id: json['id'] as String,
    name: json['name'] as String,
    slug: json['slug'] as String,
    sortOrder: (json['sort_order'] as num?)?.toInt() ?? 0,
    ageRestricted: json['age_restricted'] as bool? ?? false,
    noReturn: json['no_return'] as bool? ?? false,
    children:
        (json['children'] as List<dynamic>?)
            ?.map((c) => Category.fromJson(c as Map<String, dynamic>))
            .toList() ??
        const [],
  );
}
